sympy/solvers/solveset.py

```diff
@@ -790,7 +791,7 @@ def _solveset(f, symbol, domain, _check=False):
     from sympy.simplify.simplify import signsimp
 
     orig_f = f
-    tf = f = together(f)
+    tf = together(f)
     if f.is_Mul:
```

Comment:
This should now be 

```
if tf.is_Mul:
    coeff, f = tf.as_independent(symbol, as_Add=False)
    if coeff...
        f = tf
```

Reply:
By doing this we are in a way doing the same thing. We are checking the combined form of the eq (`(-y*exp(18000000*x) + exp(18000000*x) - 1)*exp(-18000000*x)`).

Reply:
ok, then let's remove the tf calculation and just write `f = together(f)` in the `if coeff...` block.

Reply:
This takes care of the test, too:

```diff
diff --git a/sympy/solvers/solveset.py b/sympy/solvers/solveset.py
index ce6074f..693acff 100644
--- a/sympy/solvers/solveset.py
+++ b/sympy/solvers/solveset.py
@@ -685,8 +685,8 @@ def _solve_abs(f, symbol, domain):
             complex domain.'''))
     p, q, r = Wild('p'), Wild('q'), Wild('r')
     pattern_match = f.match(p*Abs(q) + r) or {}
-    if not pattern_match.get(p, S.Zero).is_zero:
-        f_p, f_q, f_r = pattern_match[p], pattern_match[q], pattern_match[r]
+    f_p, f_q, f_r = [pattern_match.get(i, S.Zero) for i in (p, q, r)]
+    if not (f_p.is_zero or f_q.is_zero):

         domain = continuous_domain(f_q, symbol, domain)
         q_pos_cond = solve_univariate_inequality(f_q >= 0, symbol,
@@ -787,12 +787,12 @@ def _solveset(f, symbol, domain, _check=False):
     from sympy.simplify.simplify import signsimp

     orig_f = f
-    tf = f = together(f)
     if f.is_Mul:
         coeff, f = f.as_independent(symbol, as_Add=False)
         if coeff in set([S.ComplexInfinity, S.NegativeInfinity, S.Infinity]):
-            f = tf
-    if f.is_Add:
+            f = together(f)
+    elif f.is_Add:
         a, h = f.as_independent(symbol)
         m, h = h.as_independent(symbol, as_Add=False)
         if m not in set([S.ComplexInfinity, S.Zero, S.Infinity,

```